### PR TITLE
Hosting Command Palette: Add banner on /sites

### PR DESCRIPTION
**Pull request by sejas**
- Related to https://github.com/Automattic/dotcom-forge/issues/4409

## Proposed Changes

* Affordance banner to promote the Hosting Command Palette on `/sites`

## Testing Instructions

Mobile

* Navigate to `/sites` on a mobile phone
* Observe there is no new banner

Reduce motion

* Navigate to `/sites` on Desktop
* Set your Mac `Preferences > Accessibility > Display > Reduce motion` toggle enabled.
* Observe there is no animation
* Set your Mac `Preferences > Accessibility > Display > Reduce motion` toggle disabled.
* Observe there is animation

Happy path desktop

* Navigate to `/sites` on your desktop
* Observe the new banner. On Mac it displays `⌘K` on other platforms `Ctrl+K`
* Click on the X button to close the banner
* Recharge the page and observe the banner doesn't appear anymore

## Screenshots

Mac:
Screenshot 2023-12-20 at 19 22 36

Other platforms:
Screenshot 2023-12-20 at 19 23 24

"Hidden" on touchable screens:

Screenshot 2023-12-20 at 20 54 44

## Screencast

https://github.com/Automattic/wp-calypso/assets/779993/8d1c5d5a-b7da-44f8-94dc-4557ac9bd4d9

https://github.com/Automattic/wp-calypso/assets/779993/55b6529c-8ba7-41ad-b900-160921b8ff55

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by sejas** [comment]
We will not merge this PR until January, when we are back from our AFK.

**Comment by javierarce** [comment]
This looks pretty nice! Some comments:

- Could we add `user-select: none` to the animation so the text cannot be highlighted?
- If I resize the screen and make it small, the banner breaks. I think we should hide the animation when the banner is small. We could make it disappear before the text spans three lines.
- Finally, the icon at the beginning of the banner needs some tweaking in terms of border-width (1px) and border-radius (2px) Also, the spacing between the shortcut and the text is a bit too large.

For reference:

image

**Comment by sejas** [comment]
@javierarce, thanks for the feedback, I applied the changes, could you take another look?

> - Could we add `user-select: none` to the animation so the text cannot be highlighted?

Done! https://github.com/Automattic/wp-calypso/pull/85564/commits/23cd6caf41c2bec441c3d84502dfce406097af88

> - If I resize the screen and make it small, the banner breaks. I think we should hide the animation when the banner is small. We could make it disappear before the text spans three lines.

Done! https://github.com/Automattic/wp-calypso/pull/85564/commits/62a01d947c8389ed865f31eecdfed42e98f2f20d

https://github.com/Automattic/wp-calypso/assets/779993/b9144478-4807-42c3-a3ba-1be9bb3028e7

> - Finally, the icon at the beginning of the banner needs some tweaking in terms of border-width (1px) and border-radius (2px) Also, the spacing between the shortcut and the text is a bit too large.

Done! https://github.com/Automattic/wp-calypso/pull/85564/commits/df510ce6d2630129a67d5cdcdc632ac311f35b36

The spacings are easy to fix together. The first iteration I used a transparent PNG to nail them, because I cannot use the same numeric distances from figma.

**Comment by wojtekn** [comment]
I tested it, and in general, it works as expected. Some thoughts below.

The banner feels crowded in smaller screen sizes. Should we hide it based on the breakpoint and not the touch availability?

Screenshot 2024-01-09 at 09 28 51

Adding more to check based on touch availability - is it correct to hide the banner if the device has a touch feature? How would it work for touchscreen laptops or tablets with connected keyboards?

I wonder if the animation is clear enough. It took me some time to understand what it shows, and my first reaction was to click those elements. I thought those were buttons that were supposed to open the command palette.

**Comment by katinthehatsite** [comment]
I like the animation that changes the commands in the banner - nice work!

I agree with Wojtek - but perhaps not only for the smaller screens. I feel like the banner generally feels a bit crowded. Perhaps a smaller font size or something that would reduce the "boxiness" of the banner. I feel like there are too many borders in conjunction with other elements that make the design a bit crowded - there is a  square border on the command name, a  square border of the banner itself, a square border on the buttons: 

Screenshot 2024-01-09 at 10 14 14 AM

File: client/sites-dashboard/components/hosting-command-palette-banner.tsx
Comment (katinthehatsite) [inline client/sites-dashboard/components/hosting-command-palette-banner.tsx:225]: I feel like we need either `.` or `!` here. I did look at other implementations though and they don't seem to have anything at the end so perhaps that might be fine for consistency with other implementations (example: https://github.com/Automattic/wp-calypso/blob/30f0baae2fff09b2580c3a7af7c3fbc80491253f/client/my-sites/stats/stats-notices/tier-upgrade-notice.tsx#L50 )

**Comment by katinthehatsite** [comment]
I was playing with the sizing a bit on my end - perhaps dropping the icon size to a bit smaller one on desktop could make it feel a bit less boxy.

This is the current look:

Screenshot 2024-01-09 at 10 14 14 AM

With the smaller size:

Screenshot 2024-01-09 at 10 24 19 AM

With the smaller icon and text on mobile:

Screenshot 2024-01-09 at 10 29 20 AM

**Comment by javierarce** [comment]
> Should we hide it based on the breakpoint and not the touch availability?

I'd do this. I think that's the reason I didn't design a version for smaller screens. Other less drastic solution that could make the banner look better on smaller screens would be to also hide the left icon.

**Comment by javierarce** [comment]
> I wonder if the animation is clear enough. It took me some time to understand what it shows, and my first reaction was to click those elements. I thought those were buttons that were supposed to open the command palette.

I've heard another person expressing the same worry… so let me try to make some tweaks to the design and see if we can improve the banner and avoid possible confusions.

**Comment by javierarce** [comment]
What do you think of this idea, @wojtekn (and everyone)? Instead of using a border, we could remove it and add a shadow. This change would make the item design more similar to its display in the palette and distinguish it from a regular button.

https://github.com/Automattic/wp-calypso/assets/4933/9c130a09-533b-45e7-8c1c-121315228f9a

**Comment by wojtekn** [comment]
> What do you think of this idea, @wojtekn (and everyone)? Instead of using a border, we could remove it and add a shadow. This change would make the item design more similar to its display in the palette and distinguish it from a regular button.

@javierarce  I think it's better now. What if we added a click handler to open the pre-filled command palette when the user clicks any of the states?

**Comment by sejas** [comment]
> @javierarce I think it's better now. What if we added a click handler to open the pre-filled command palette when the user clicks any of the states?

I was thinking about this, although the main goal is to teach users to press the keyboard shortcuts, and it would expose the command palette for mobile devices if we hide the banner based on the breakpoint.

> Adding more to check based on touch availability - is it correct to hide the banner if the device has a touch feature? How would it work for touchscreen laptops or tablets with connected keyboards?

Here is a bit more context: p1702546774835439/1702380294.132179-slack-C9EJ7KSGH . There is no perfect solution. I guess using the breakpoint is better, because a person from a tablet could switch to desktop to try the feature.

**Comment by sejas** [comment]
I improved the spacing, base the visibility on breakpoints instead of touchable screens, and I added the blue background gradient. This is how it looks:



Screenshot 2024-01-09 at 17 02 14

https://github.com/Automattic/wp-calypso/assets/779993/3ef3cef6-ad41-40b1-8858-5c989887fe8f


Is it good to merge it?

**Comment by wojtekn** [comment]
> I was thinking about this, although the main goal is to teach users to press the keyboard shortcuts, and it would expose the command palette for mobile devices if we hide the banner based on the breakpoint.

That sounds fair. Thanks for providing more context on the decision.

**Review by wojtekn — APPROVED** [review]

**Review by katinthehatsite — APPROVED** [review]
I like the look with the shadow much better! Nice work!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10819516

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @sejas for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10819516).

**Pull request merged**
